didcomm_resolver/role/requester/routes.py

```diff
@@ -160,7 +160,7 @@ async def connection(request: web.BaseRequest):
         metadata = await record.metadata_get(session, METADATA_KEY)
         if not isinstance(metadata, dict):
             metadata = json.loads(metadata.value)
-        methods = [] if metadata is None else metadata["methods"]
+        methods = set() if metadata is None else metadata["methods"]
```

Comment:
Same as previous comment. We need something like the following to actually coerce this value into a set:
```suggestion
        methods = set() if metadata is None else set(metadata["methods"])
```